packages/core/src/beacon.ts

```diff
@@ -0,0 +1,20 @@
+import { getImplementationAddressFromBeacon } from '.';
+import { InvalidBeaconImplementation } from './impl-address';
+import { EthereumProvider } from './provider';
+
+/**
+ * Checks if the address looks like a beacon.
+ *
+ * @returns true if the address has an implementation() function that returns an address, false otherwise.
+ */
+export async function isBeacon(provider: EthereumProvider, beaconAddress: string) {
+  try {
+    return await getImplementationAddressFromBeacon(provider, beaconAddress);
```

Comment:
This is just returning the string, some comparison to get a boolean is missing I believe.